Update optimum-habana-perf.md using Gaudi-Tutorial

louie-tsai opened the pull request:
direct users to use Dockerfile for OH benchmarking and also use Benchmark script directly

jedwards-habana commented on AEM/optimum-habana-perf.md:
```diff
@@ -1,48 +1,57 @@
+## Build and Run the Benchmark Docker instance 
+
+### Get Dockerfile and Benchmark scripts
+First, get the Dockerfile and Benchmark scripts from Gaudi-Tutorial GitHub Repositroy following below command.  
```
Grammar and spelling improvements for sentence. Change to: "First, obtain the Dockerfile and benchmark scripts from the Gaudi-Tutorial GitHub repository using the command below."

jedwards-habana commented on AEM/optimum-habana-perf.md:
```diff
@@ -1,48 +1,57 @@
-### Install required packages inside docker
-First, install the optimum-habana:
+### Docker Build
+To build the image from the Dockerfile, please follow below command to build the optimum-habana-text-gen image.
```
Improvements: "To build the image from the Dockerfile, please use the command below to create the optimum-habana-text-gen image."

jedwards-habana commented on AEM/optimum-habana-perf.md:
```diff
@@ -1,48 +1,57 @@
-
-Second, install the requirements:
+### Docker Run
+After docker build, users could follow below command to run and docker instance and users will be in the docker instance under text-generation folder.
```
Change to: "After building the Docker image, users can use the command below to run a Docker instance, which will place them in the text-generation folder within the instance."

jedwards-habana commented on AEM/optimum-habana-perf.md:
````diff
@@ -1,48 +1,57 @@
+docker run -it --runtime=habana -e HABANA_VISIBLE_DEVICES=all -e OMPI_MCA_btl_vader_single_copy_mechanism=none   --cap-add=ALL --privileged=true  --net=host --ipc=host optimum-habana-text-gen:latest
 ```
+> [!NOTE]
+> The Huggingface model file size might be large, so we recommend to use an external disk as Huggingface hub folder. \
````
Change "so we recommend to use an external disk as Huggingface hub folder." to "using an external disk to house the Huggingface hub folder is recommended."

jedwards-habana commented on AEM/optimum-habana-perf.md:
````diff
@@ -1,48 +1,57 @@
 ```
+> [!NOTE]
+> The Huggingface model file size might be large, so we recommend to use an external disk as Huggingface hub folder. \
+> Please export HF_HOME environment variable to your external disk and then export the mount point into docker instance. \
````
What are you trying to say here? Should users pass the HF_HOME environment variable to their docker file using the -e option and then mount the path to the external disk into that container instance?

louie-tsai replied:
yes. also changed the contents to explain it better.

jedwards-habana commented on AEM/optimum-habana-perf.md:
```diff
@@ -1,48 +1,57 @@
+> ex: "-e HF_HOME=/mnt/huggingface -v /mnt:/mnt"
+
+# Run Benchmark with Benchmark.py
+Benchmark script will run all the models with different input len, output len and batch size and generate a report to compare all published numbers in [Gaudi Model Performance](https://www.intel.com/content/www/us/en/developer/platform/gaudi/model-performance.html).  
```
"The benchmark script"

jedwards-habana commented on AEM/optimum-habana-perf.md:
```diff
@@ -1,48 +1,57 @@
 
-For `run_lm_eval.py`:
+## Gaudi3
+Different json file are provided for different Gaudi Software version like 1.19 and 1.20 on Gaudi3.
```
Change "file" to "files"; change "version" to "versions". You may want to change "Gaudi Software" to "Intel Gaudi software", and place the correct copyright symbols on the Intel and the Gaudi.

jedwards-habana commented on AEM/optimum-habana-perf.md:
```diff
@@ -1,48 +1,57 @@
-
-Then, to use [DeepSpeed-inference](https://docs.habana.ai/en/latest/PyTorch/DeepSpeed/Inference_Using_DeepSpeed.html), install DeepSpeed as follows: 
+## Gaudi2
+To do benchmarking on a machine with 8 Gaudi2 cards, just run the below command instead inside the docker instance. 
```
Change, "To do benchmarking on a machine with 8 Gaudi2 cards, just run the below command instead inside the docker instance." to "To do benchmarking on a machine with 8 Gaudi2 cards, run the same Benchmark.py command but specify the Gaudi version using the GAUDI_VER environment variable."

jedwards-habana commented on AEM/optimum-habana-perf.md:
```diff
@@ -1,48 +1,57 @@
-
-# Tensor quantization statisics measurement
+## HTML Report
+A html report will be generated under a folder with timestamp.  
```
Change to "An HTML report will be generated in a timestamped folder created at the time of execution.

jedwards-habana commented on AEM/optimum-habana-perf.md:
```diff
@@ -1,48 +1,57 @@
-# Tensor quantization statisics measurement
+## HTML Report
+A html report will be generated under a folder with timestamp.  
+The [html report](https://github.com/HabanaAI/Gaudi-tutorials/tree/main/PyTorch/Hugging_Face_pipelines/Benchmarking_on_Optimum-habana_with_fp8#html-report) also has a perf_ratio column to compare the measure numbers with reference perf numbers. 
```
What are "measure numbers"? Are these previously generated numbers?

louie-tsai replied:
I called previously generated numbers as reference perf numbers here, so measured numbers are from this benchmark run.  let me know any modification needed.

jedwards-habana replied:
Let's use "previous benchmark results".

louie-tsai replied:
addressed it accordingly.

jedwards-habana reviewed (changes requested):
Most of my comments are about sentence structure and spelling changes.

louie-tsai commented:
> Most of my comments are about sentence structure and spelling changes.

much appreciated for your prompt response and addressed all comments accordingly.

jedwards-habana reviewed (approved):
LGTM
